tests/components/filesize/test_sensor.py

```diff
@@ -21,40 +21,36 @@ def create_file(path):
         test_file.write("test")
 
 
-class TestFileSensor(unittest.TestCase):
-    """Test the filesize sensor."""
+@pytest.fixture(autouse=True)
+def remove_file():
+    """Remove test file."""
+    if os.path.isfile(TEST_FILE):
+        os.remove(TEST_FILE)
 
-    def setup_method(self, method):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.hass.config.allowlist_external_dirs = {TEST_DIR}
 
-    def teardown_method(self, method):
-        """Stop everything that was started."""
-        if os.path.isfile(TEST_FILE):
-            os.remove(TEST_FILE)
-        self.hass.stop()
+async def test_invalid_path(hass):
+    """Test that an invalid path is caught."""
+    config = {"sensor": {"platform": "filesize", CONF_FILE_PATHS: ["invalid_path"]}}
+    assert await async_setup_component(hass, "sensor", config)
+    await hass.async_block_till_done()
+    assert len(hass.states.async_entity_ids()) == 0
 
-    def test_invalid_path(self):
-        """Test that an invalid path is caught."""
-        config = {"sensor": {"platform": "filesize", CONF_FILE_PATHS: ["invalid_path"]}}
-        assert setup_component(self.hass, "sensor", config)
-        assert len(self.hass.states.entity_ids()) == 0
 
-    def test_valid_path(self):
-        """Test for a valid path."""
-        create_file(TEST_FILE)
-        config = {"sensor": {"platform": "filesize", CONF_FILE_PATHS: [TEST_FILE]}}
-        assert setup_component(self.hass, "sensor", config)
-        self.hass.block_till_done()
-        assert len(self.hass.states.entity_ids()) == 1
-        state = self.hass.states.get("sensor.mock_file_test_filesize_txt")
-        assert state.state == "0.0"
-        assert state.attributes.get("bytes") == 4
+async def test_valid_path(hass):
+    """Test for a valid path."""
+    create_file(TEST_FILE)
+    config = {"sensor": {"platform": "filesize", CONF_FILE_PATHS: [TEST_FILE]}}
+    hass.config.allowlist_external_dirs = {TEST_DIR}
+    assert await async_setup_component(hass, "sensor", config)
+    await hass.async_block_till_done()
+    assert len(hass.states.async_entity_ids()) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)